assets/themes/hailhydrant/hailhydrant.json

```diff
@@ -0,0 +1,601 @@
+                "en": "<img style=\"width:60px; height:50px\" src=\"https://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Berlin_hydrant_20050211_p1000517.jpg/320px-Berlin_hydrant_20050211_p1000517.jpg\" /> Underground type."
+              }
+            }
+          ]
+        },
+        {
+          "question": {
+            "en": "How many couplings are there?"
+          },
+          "render": {
+            "en": "Couplings: {couplings}"
+          },
+          "freeform": {
+            "key": "couplings",
+            "type": "pnat"
+          },
+          "mappings": [
+            {
+              "if": {
+                "and": [
+                  "couplings=1"
+                ]
+              },
+              "then": {
+                "en": "The hydrant has one (1) coupling."
+              }
+            },
+            {
+              "if": {
+                "and": [
+                  "couplings=2"
+                ]
+              },
+              "then": {
+                "en": "The hydrant has two (2) couplings."
+              }
+            },
+            {
+              "if": {
+                "and": [
+                  "couplings=3"
+                ]
+              },
+              "then": {
+                "en": "The hydrant has three (3) coupling."
+              }
+            }
+          ]
+        },
+        {
+          "question": {
+            "en": "Update the lifecycle status of the hydrant."
+          },
+          "render": {
+            "en": "Lifecycle status"
+          },
+          "freeform": {
+            "key": "disused:emergency"
+          },
+          "mappings": [
+            {
+              "if": {
+                "and": [
+                  "emergency=fire_hydrant"
+                ]
+              },
+              "then": {
+                "en": "Hydrant is (fully or partially) working."
+              }
+            },
+            {
+              "if": {
+                "and": [
+                  "disused:emergency=fire_hydrant",
+                  "emergency="
+                ]
+              },
+              "then": {
+                "en": "Hydrant is unavailable for use."
+              }
+            },
+            {
+              "if": {
+                "and": [
+                  "removed:emergency=fire_hydrant",
+                  "emergency="
+                ]
+              },
+              "then": {
+                "en": "Hydrant has been removed."
+              }
+            }
+          ]
+        },
+        {
+          "question": {
+            "en": "Describe the position of the hydrant."
+          },
+          "render": {
+            "en": "Position: {fire_hydrant:position}"
+          },
+          "mappings": [
+            {
+              "if": {
+                "and": [
+                  "fire_hydrant:position=sidewalk"
+                ]
+              },
+              "then": {
+                "en": "Sidewalk"
```

Comment:
This is the cause of your "weird rendering". The mapping here sees `fire_hydrant:position=sidewalk", so it will show "sidewalk" in the popup.

The 'render'-key is only used if no mapping matches.

Reply:
Also note that it might be a good idea to write out this option a bit more, as `This hydrant is located on <b>the sidewalk</b>`, instead of just writing `Sidewalk`. _Every mapping must be clear to read without any further context_